jdbc/src/test/org/apache/hive/jdbc/TestHiveBaseResultSet.java

```diff
@@ -240,6 +245,56 @@ public void testGetBooleanStringTrue() throws SQLException {
     Assert.assertFalse(resultSet.wasNull());
   }
 
+  /**
+   * HIVE-28358 getClob(int) != null
+   */
+  @Test
+  public void testGetClobString() throws SQLException, IOException {
+    FieldSchema fieldSchema = new FieldSchema();
+    fieldSchema.setType("varchar(64)");
+
+    List<FieldSchema> fieldSchemas = Arrays.asList(fieldSchema);
+    TableSchema schema = new TableSchema(fieldSchemas);
+
+    HiveBaseResultSet resultSet = Mockito.mock(HiveBaseResultSet.class);
+    resultSet.row = new Object[] {"ABC"};
+
+    when(resultSet.getSchema()).thenReturn(schema);
+    when(resultSet.getString(1)).thenCallRealMethod();
+	when(resultSet.getClob(1)).thenCallRealMethod();
+    when(resultSet.wasNull()).thenCallRealMethod();
+
+    Clob clob = resultSet.getClob(1);
+    try (Reader clobReader = clob.getCharacterStream())
+    {
+      Assert.assertEquals("ABC", new BufferedReader(clobReader).lines().collect(Collectors.joining(System.lineSeparator())));
+    }
+    Assert.assertFalse(resultSet.wasNull());
+  }
+
+  /**
+   * HIVE-28358 getClob(int) == null 
+   */
+  @Test
+  public void testGetClobNull() throws SQLException {
+    FieldSchema fieldSchema = new FieldSchema();
+    fieldSchema.setType("varchar(64)");
+
+    List<FieldSchema> fieldSchemas = Arrays.asList(fieldSchema);
+    TableSchema schema = new TableSchema(fieldSchemas);
+
+    HiveBaseResultSet resultSet = Mockito.mock(HiveBaseResultSet.class);
```

Comment:
use spy instead and you won't need to do `thenCallRealMethod()`

Reply:
else LGTM

Reply:
changed tests to use .spy().